src/core/Target.mli

```diff
@@ -0,0 +1,144 @@
+    associated information, or anything else a Origin.t can designate.
+
+    This should be used when reporting a location to the user. *)
+  internal_path_to_content : Fpath.t;
+      (** The path to a file which contains the data to be scanned. This could be
+    the same as the origin, if the origin is a path to a regular file (or an
+    absolute path to the same), or it could be a tempfile. This should be
+    used to obtain the contents of the target, but not for reporting to the
+    user, other than possibly for debugging purposes. *)
+}
+[@@deriving show]
+(** Information about where a target from for both the purpose of
+   {ul
+    {- informing the user: [origin]}
+    {- obtaining the contents: [internal_path_to_content]}
+  } *)
+
+(* deriving eq appears to not work on mutual definitions with and. See also
+   <https://github.com/ocaml-ppx/ppx_deriving/issues/272>, for deriving make,
+   which may be relevant. Therefore, we just define it here. *)
+val equal_target_path : target_path -> target_path -> bool
+
+val code_of_origin :
+  ?lockfile:lockfile ->
+  Xlang.t ->
+  Semgrep_output_v1_t.product list ->
+  Origin.t ->
+  code
+(** [code_of_origin analyzer products origin] is the target
+      location for a source code target originating from [origin] to be
+      analyzed with [analyzer] for [products]. If [lockfile] is specified then
+      it shall be used as the associated lockfile if dependency patterns are
+      to be ran.
+
+      This function should be generally preferred over creating a record
+      directly, since it can peform actions which may be required when creating
+      a target from certain types of origins, such as generating a tempfile.
+ *)
+
+val lockfile_of_origin :
+  ?manifest:manifest -> Lockfile_kind.t -> Origin.t -> lockfile
+(** [lockfile_of_origin k origin] is the target
+      location for a lockfile target originating from [origin] of kind [k].
+      If [manifest] is specified, it shall be used as the associated manifest.
+
+      This function should be generally preferred over creating a record
+      directly, since it can peform actions which may be required when creating
+      a target from certain types of origins, such as generating a tempfile.
+ *)
+
+val manifest_of_origin : Manifest_kind.t -> Origin.t -> manifest
+(** [manifest_of_origin k origin] is the target
+      location for a manifest target originating from [origin] of kind [k].
+
+      This function should be generally preferred over creating a record
+      directly, since it can peform actions which may be required when creating
+      a target from certain types of origins, such as generating a tempfile.
+ *)
+
+val internal_path_to_content : t -> Fpath.t
+(** [internal_path_to_content target] is the path to a file containing the
+    contents of [target]. *)
+
+val origin : t -> Origin.t
+(** [origin target] is the user-reportable origin of [target]. *)
```

Comment:
what are those empty Target_location.ml files below?

Reply:
Cleaned up in 8c905bd51f56312325aaa7761cfc80868b953845